src/Shared/LogMessagePacketBase.cs

```diff
@@ -573,6 +641,15 @@ private LoggingEventType GetLoggingEventId(BuildEventArgs eventArg)
             {
                 return LoggingEventType.ProjectStartedEvent;
             }
+            else if (eventType == typeof(ExternalProjectStartedEventArgs))
```

Comment:
I copied used coding practice above. I am not sure if introducing `LoggingEventType` into events is desirable, as it would enforce events to include de/ser implementation details. Creating dictionary for mapping `Type -> LoggingEventType` would be better,

Reply:
Sounds good, thank you for checking.